docs/Compatibility.md

```diff
@@ -4,156 +4,259 @@ The Ruby Datadog Trace library is open source. See the [dd-trace-rb][1] GitHub r
+|       |                            | 2.4       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.3       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.2       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.1       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.0       | [EOL](#support-eol) | < 0.50.0            |
+|       |                            | 1.9.3     | [EOL](#support-eol) | < 0.27.0            |
+|       |                            | 1.9.1     | [EOL](#support-eol) | < 0.27.0            |
+| JRuby | https://www.jruby.org      | 9.3       | [GA](#support-ga)   | Latest              |
+|       |                            | 9.2.21.0+ | [GA](#support-ga)   | Latest              |
 
 ### Supported web servers
 
-| Type      | Documentation                     | Version      | Support type |
-| --------- | --------------------------------- | ------------ | ------------ |
-| Puma      | http://puma.io/                   | 2.16+ / 3.6+ | Full         |
-| Unicorn   | https://bogomips.org/unicorn/     | 4.8+ / 5.1+  | Full         |
-| Passenger | https://www.phusionpassenger.com/ | 5.0+         | Full         |
+| Type      | Documentation                     | Version      | Support type      |
+|-----------|-----------------------------------|--------------|-------------------|
+| Puma      | http://puma.io/                   | 2.16+ / 3.6+ | [GA](#support-ga) |
+| Unicorn   | https://bogomips.org/unicorn/     | 4.8+ / 5.1+  | [GA](#support-ga) |
+| Passenger | https://www.phusionpassenger.com/ | 5.0+         | [GA](#support-ga) |
 
 ### Supported tracing frameworks
 
-| Type        | Documentation                                   | Version               | Gem version support |
-| ----------- | ----------------------------------------------- | --------------------- | ------------------- |
-| OpenTracing | https://github.com/opentracing/opentracing-ruby | 0.4.1+                | < 2.0.0            |
+| Type        | Documentation                                   | Version | Support type        | Gem version support |
+|-------------|-------------------------------------------------|---------|---------------------|---------------------|
+| OpenTracing | https://github.com/opentracing/opentracing-ruby | 0.4.1+  | [EOL](#support-eol) | < 2.0.0             |
 
-*Full* support indicates all tracer features are available.
+### Supported operating systems
 
-*Deprecated* indicates support will transition to *Maintenance* in a future release.
+| OS           | Support type                        | Package version |
+|--------------|-------------------------------------|-----------------|
+| Linux x86-64 | [GA](#support-ga)                   | Latest          |
+| Linux ARM64  | [GA](#support-ga)                   | Latest          |
```

Comment:
Should we add some statement regarding Linux on other architectures? (Dev environments only? Talk to us?)

Reply:
I don't think we should mark any architectures as dev-only if we don't have some sort of CI in place for it. I could add a sentence below the table that says:

Need linux support for a CPU architecture not listed? [Contact our customer support team for special requests.](https://www.datadoghq.com/support)

Reply:
Seems reasonable!

Reply:
I don't know what the status for profiling/libdatadog is, but for libddwaf it goes like this:

a) We have binaries for supported platofrms
b) `libddwaf` is able to be compiled on basically any platform, the `ruby` platform gem can† do that.

If customers figure out a way to have it build (which is essentially having a C++ compiler + Cmake) then they can have ASM work on ppc64le, RISC-V, FreeBSD, Solaris, what have you.

† It's not done yet but I'm planning to do it to support the people that use `force_ruby_platform` on supported platforms. The above will come as a consequence of that.

Reply:
> I don't know what the status for profiling/libdatadog is, but for libddwaf it goes like this:

The current status is we're shipping libdatadog binaries for [x86-64, arm64] x [linux glibc, linux musl], so that's what profiler supports out of the box.

(It's undocumented, but if you can manually compile libdatadog to more platforms, and profiler does usually work on them as well).

---

Having said that, @lloeki I think @ekump has a reasonable point about the difference between "we have binaries for obscure platforms" and "we're actively testing and supporting them". If we're not testing on Solaris, and would struggle if someone came to us with a bug that only triggered on Solaris, should we state we *support* Solaris?

I think it's OK to have a "best-effort, come talk to us if this really important to you" tier, but I think it makes sense to be clear in this table about how much effort we would spend on issues for these platforms.

Reply:
> If we're not testing on Solaris, and would struggle if someone came to us with a bug that only triggered on Solaris, should we state we support Solaris?

I agree, hence why I said these fall under "unsupported".

The conundrum I have is that once I add support for the `ruby` platform to properly support `{x86_64,aarch64}-linux-{gnu,musl}` the *other* platforms are going to start working but we don't want ambiguity as to whether they're supported through the fact that the `ruby` platform is.

> I think it's OK to have a "best-effort, come talk to us if this really important to you" tier, but I think it makes sense to be clear in this table about how much effort we would spend on issues for these platforms.

I like the "best effort" angle, especially if it encourages external contributions. And I agree, this is all about clarity in communicating the effort we are able to expend (both in terms of human resources e.g dev/debug time and technical resources e.g CI).